packages/core/src/external/sde/command/bundle/get-bundles.ts

```diff
@@ -0,0 +1,67 @@
+import _ from "lodash";
+import { S3Utils } from "../../../aws/s3";
+import { Config } from "../../../../util/config";
+import { Bundle, BundleEntry } from "@medplum/fhirtypes";
+import { getDataExtractionFilePrefix } from "../../file-names";
+import { executeAsynchronously } from "../../../../util/concurrency";
+import { parseFhirBundle } from "@metriport/shared/medical";
+import { out } from "../../../../util/log";
+
+const numberOfParallelExecutions = 10;
+
+interface GetBundlesParams {
+  cxId: string;
+  patientId: string;
+}
+
+/**
+ * Retrieves all bundles that have been extracted for a particular cxId and patientId. Will return an
+ * empty array if no bundles have been extracted, or the environment is not configured for SDE.
+ *
+ * @param cxId - The CX ID.
+ * @param patientId - The patient ID.
+ * @returns All bundles that have been extracted for the patient.
+ */
+export async function getBundles({ cxId, patientId }: GetBundlesParams): Promise<Bundle[]> {
+  const { log } = out("sde.getBundles");
+  const bucketName = Config.getStructuredDataBucketName();
+  if (!bucketName) {
+    log(`No structured data bucket name found, skipping`);
+    return [];
+  }
```

Comment:
Shouldn't we build this in a way that's available in all environments? The matter of whether to enable it to `sandbox` or not is something we concentrate in one place, but everywhere else runs w/o knowing about this.

Reply:
In the config the bucket is optional to avoid past issues with the required bucket name, so for that reason I assumed that the result from the environment variable should also be optional.